src/game/Spell.cpp

```diff
@@ -5100,9 +5100,7 @@ SpellCastResult Spell::CheckCast(bool strict)
                     else
                     {
                         Pet* dbPet = new Pet;
-                        if (dbPet->LoadPetFromDB((Player*)m_caster, 0))
-                            return SPELL_CAST_OK;           // still returns an error to the player, so this error must come from somewhere else...
-                        else
+                        if (!dbPet->TryLoadFromDB((Player*)m_caster, 0))
```

Comment:
if there is something to load, we continue to SummonPet, which is fine, but this means that the check in summonPet is redundant, because if the hunter has nothing to load, then he will return spell failed instead and never go through to SummonPet -> this should also make the whole skip bool redundant.